File: clients/ios/Lextures/Core/LMS/LMSFeatureModelsGroups.swift
import Foundation

// MARK: - Group spaces & collab docs (M7.4)

struct GroupPublic: Codable, Identifiable, Hashable {
    var id: String
    var groupSetId: String
    var name: String
    var sortOrder: Int
    var createdAt: String
    var memberCount: Int
}

struct GroupsListResponse: Decodable {
    var groups: [GroupPublic]?
}

enum CollabDocType: String, Codable, Hashable {
    case richText = "rich_text"
    case whiteboard
}

struct CollabDoc: Codable, Identifiable, Hashable {
    var id: String
    var courseId: String
    var groupId: String?
    var title: String
    var docType: CollabDocType
    var createdBy: String
    var createdAt: String
    var updatedAt: String
}

struct CollabDocsListResponse: Decodable {
    var docs: [CollabDoc]?
}

struct CollabDocSnapshot: Codable, Identifiable, Hashable {
    var id: String
    var docId: String
    var authorId: String
    var takenAt: String
}

struct CollabDocSnapshotsResponse: Decodable {
    var snapshots: [CollabDocSnapshot]?
}

struct GroupFeedContext: Hashable {
    var groupId: String
    var groupName: String
}

struct GroupChannelRoute: Hashable {
    var groupId: String
    var groupName: String
    var channelId: String
    var channelName: String
}

struct GroupSpaceRoute: Hashable {
    var group: GroupPublic
}

struct CollabDocRoute: Hashable {
    var docId: String
    var title: String
}

File: clients/android/app/src/main/kotlin/com/lextures/android/core/lms/LmsFeatureModelsGroups.kt
package com.lextures.android.core.lms

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class GroupPublic(
    val id: String,
    val groupSetId: String,
    val name: String,
    val sortOrder: Int = 0,
    val createdAt: String = "",
    val memberCount: Int = 0,
)

@Serializable
data class GroupsListResponse(
    val groups: List<GroupPublic> = emptyList(),
)

@Serializable
enum class CollabDocType {
    @SerialName("rich_text") RichText,
    @SerialName("whiteboard") Whiteboard,
}

@Serializable
data class CollabDoc(
    val id: String,
    val courseId: String,
    val groupId: String? = null,
    val title: String,
    val docType: CollabDocType = CollabDocType.RichText,
    val createdBy: String = "",
    val createdAt: String = "",
    val updatedAt: String = "",
)

@Serializable
data class CollabDocsListResponse(
    val docs: List<CollabDoc> = emptyList(),
)

data class GroupFeedContext(
    val groupId: String,
    val groupName: String,
)

data class GroupMemberRow(
    val id: String,
    val displayName: String,
    val email: String,
)

enum class GroupSpaceTab {
    Members,
    Discussion,
    Files,
    Docs,
}